arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArcGISArView.kt

```diff
@@ -391,40 +587,52 @@ final class ArcGISArView : FrameLayout, DefaultLifecycleObserver, Scene.OnUpdate
+            if (isTracking) {
+                Quaternion.multiply(
+                    compensationQuaternion,
+                    Quaternion(
+                        arCamera.displayOrientedPose.rotationQuaternion[0],
+                        arCamera.displayOrientedPose.rotationQuaternion[1],
+                        arCamera.displayOrientedPose.rotationQuaternion[2],
+                        arCamera.displayOrientedPose.rotationQuaternion[3]
+                    )
+                ).let { compensatedQuaternion ->
+                    // create a Pair from the rotation quaternion and translation to create a TransformationMatrix
+                    Pair(
+                        compensatedQuaternion,
+                        arCamera.displayOrientedPose.translation.map { it.toDouble() }.toDoubleArray()
+                    )
+                }.let {
+                    // we must flip the y and z co-ordinates due to the compensation quaternion
                     TransformationMatrix.createWithQuaternionAndTranslation(
-                        it.first[0],
-                        it.first[1],
-                        it.first[2],
-                        it.first[3],
+                        it.first.x.toDouble(),
+                        it.first.y.toDouble(),
+                        it.first.z.toDouble(),
+                        it.first.w.toDouble(),
                         it.second[0],
-                        it.second[1],
-                        it.second[2]
+                        -it.second[2],
+                        it.second[1]
                     )
                 }.let { arCoreTransMatrix ->
                     cameraController.transformationMatrix =
                         initialTransformationMatrix.addTransformation(arCoreTransMatrix)
                 }
-            }
-            arCamera.imageIntrinsics.let {
-                sceneView.setFieldOfViewFromLensIntrinsics(
-                    it.focalLength[0],
-                    it.focalLength[1],
-                    it.principalPoint[0],
-                    it.principalPoint[1],
-                    it.imageDimensions[0].toFloat(),
-                    it.imageDimensions[1].toFloat(),
-                    deviceOrientation
-                )
-            }
-            if (sceneView.isManualRenderingEnabled) {
-                sceneView.renderFrame()
+
+                arCamera.imageIntrinsics.let {
+                    sceneView.setFieldOfViewFromLensIntrinsics(
+                        it.focalLength[0],
+                        it.focalLength[1],
+                        it.principalPoint[0],
+                        it.principalPoint[1],
+                        it.imageDimensions[0].toFloat(),
+                        it.imageDimensions[1].toFloat(),
+                        deviceOrientation
+                    )
+                }
+                if (isUsingARCore && sceneView.isManualRenderingEnabled) {
```

Comment:
As I mentioned in our call, because the user has access to `sceneView`, they can change `isManualRenderingEnabled` themselves, this is just for safety.